base/compiler/ssair/inlining.jl

```diff
@@ -244,7 +245,7 @@ function cfg_inline_unionsplit!(ir::IRCode, idx::Int, union_split::UnionSplit,
         end
     end
     # The edge from the fallback block.
-    fully_covered || push!(from_bbs, length(state.new_cfg_blocks))
+    !handled_all_cases && push!(from_bbs, length(state.new_cfg_blocks))
```

Comment:
```suggestion
    # NOTE In the cases of `!fully_covered`, since no additional edge is required since
    #      we will inline `Core.throw_methoderror` for this dispatch candidate, which is 
    #      must-throw, making the subsequent code path unreachable.
    !handled_all_cases && push!(from_bbs, length(state.new_cfg_blocks))
```